src/arm/decoder.c

```diff
@@ -45,6 +47,15 @@ static const char* _armConditions[] = {
 	"nv"
 };
 
+static int _decodeImmediate(int imm, enum mDisassemblyStyle disassemblyStyle, char* buffer, int blen) {
+	switch(disassemblyStyle) {
```

Comment:
```suggestion
	switch (disassemblyStyle) {
```

Also, I meant having both the default and explicit cases separately listed, but it should probably be done with an if/else like you have in the SM83 implementation.